mlir/include/mlir/Transforms/Passes.td

```diff
@@ -586,4 +586,23 @@ def CompositeFixedPointPass : Pass<"composite-fixed-point-pass"> {
   ];
 }
 
+def SelectPass : Pass<"select-pass"> {
+  let summary = "Select pass";
+  let description = [{
+    Select pass allows to run multiple different set of passes based on
```

Comment:
Multiple different sets, makes me think of something else than switching on attribute to select pipeline.

Dynamically selects pass pipeline to run based on top level op attribute.